src/documentation.jl

```diff
@@ -255,18 +342,179 @@ See also: [`CryptoContext`](@ref), [`Ciphertext`](@ref)
+
+`degree` is a desired degree of approximation.
+
+Supported only in CKKS.
+
+See also: [`CryptoContext`](@ref), [`Ciphertext`](@ref)
+"""
+EvalSin
+
+"""
+    EvalCos(crypto_context::CryptoContext, ciphertext::Ciphertext, a::Float64, b::Float64,
+            degree::UInt32)
+
+Evaluate approximate cosine function on a given `ciphertext` using the Chebyshev approximation over
+the range ``[a,b]``. Return the resulting [`Ciphertext`](@ref). The input ciphertext needs to be
+derived from the given `crypto_context`.
+
+`a` is a lower bound of elements contained in the given `ciphertext`.
+
+`b` is an upper bound of elements contained in the given `ciphertext`.
+
+`degree` is a desired degree of approximation.
+
+Supported only in CKKS.
+
+See also: [`CryptoContext`](@ref), [`Ciphertext`](@ref)
+"""
+EvalCos
+
+"""
+    EvalLogistic(crypto_context::CryptoContext, ciphertext::Ciphertext, a::Float64, b::Float64,
+                 degree::UInt32)
+
+Evaluate approximate logistic function ``\\frac{1}{1 + \\exp{-x}}`` on a given `ciphertext` using
+the Chebyshev approximation over the range ``[a,b]``. Return the resulting [`Ciphertext`](@ref).
+The input ciphertext needs to be derived from the given `crypto_context`.
+
+`x` is an element contained in the given `ciphertext`
+                    
+`a` is a lower bound of elements contained in the given `ciphertext`.
+                    
+`b` is an upper bound of elements contained in the given `ciphertext`.
+                    
+`degree` is a desired degree of approximation.
+                    
+Supported only in CKKS.
+                    
+See also: [`CryptoContext`](@ref), [`Ciphertext`](@ref)             
+"""
+EvalLogistic
+
+"""
+    EvalDivide(crypto_context::CryptoContext, ciphertext::Ciphertext, a::Float64, b::Float64,
+               degree::UInt32)
+
+Evaluate approximate division function ``\\frac{1}{x}`` where ``x >= 1`` on a given `ciphertext`
+using the Chebyshev approximation over the range ``[a,b]``. Return the resulting [`Ciphertext`](@ref).
+The input ciphertext needs to be derived from the given `crypto_context`.
+
+`x` is an element contained in the given `ciphertext`
+                    
+`a` is a lower bound of elements contained in the given `ciphertext`.
+                    
+`b` is an upper bound of elements contained in the given `ciphertext`.
+                    
+`degree` is a desired degree of approximation.
+                    
+Supported only in CKKS.
+                    
+See also: [`CryptoContext`](@ref), [`Ciphertext`](@ref)             
+"""
+EvalDivide
+
+"""
+    EvalSumKeyGen(crypto_context::CryptoContext, private_key::PrivateKey, public_key::PublicKey = nothing)
+
+Generates the key map to be used by [`EvalSum`](@ref). `public_key` has to be set for NTRU schemes.
+
+Please refer to the OpenFHE documentation for more details.
+
+See also: [`CryptoContext`](@ref), [`PrivateKey`](@ref), [`PublicKey`](@ref), [`EvalSum`](@ref)
+"""
+EvalSumKeyGen
```

Comment:
I don't think this optional parameter will work: When wrapping the C++ functions with CxxWrap.jl, the compiler does not "see" the optional argument values but just assumes that all function arguments are required.

To make the last argument optional, you need to do something like I did for `EvalRotateKeyGen`, i.e., add an extra wrapper method to `src/convenience.jl` (and remove this one here):
https://github.com/sloede/OpenFHE.jl/blob/abc60a87ddfb4ed109317d7c7e8af23a28210ce5/src/convenience.jl#L191-L208